File: app/src/main/java/com/cristhian/gametobeat/ui/widget/GameWidget.java
package com.cristhian.gametobeat.ui.widget;

import android.appwidget.AppWidgetManager;
import android.appwidget.AppWidgetProvider;
import android.content.ComponentName;
import android.content.Context;
import android.content.Intent;
import android.widget.RemoteViews;

import com.cristhian.gametobeat.R;
import com.cristhian.gametobeat.database.Game;

import java.util.List;

public class GameWidget extends AppWidgetProvider {

    public static List<Game> mGames;

    static void updateAppWidget(Context context, AppWidgetManager appWidgetManager,
                                int appWidgetId) {

        // Construct the RemoteViews object
        RemoteViews views = new RemoteViews(context.getPackageName(), R.layout.widget_game);

        Intent intentGames = new Intent(context, GameWidgetService.class);
        views.setRemoteAdapter(R.id.lv_games, intentGames);

        // Instruct the widget manager to update the widget
        appWidgetManager.updateAppWidget(appWidgetId, views);
    }

    @Override
    public void onUpdate(Context context, AppWidgetManager appWidgetManager, int[] appWidgetIds) {
        // There may be multiple widgets active, so update all of them
        for (int appWidgetId : appWidgetIds) {
            updateAppWidget(context, appWidgetManager, appWidgetId);
        }
    }

    @Override
    public void onEnabled(Context context) {
    }

    @Override
    public void onDisabled(Context context) {
    }

    @Override
    public void onReceive(final Context context, Intent intent) {
        String action = intent.getAction();

        if (action.equals(AppWidgetManager.ACTION_APPWIDGET_UPDATE)) {
            AppWidgetManager appWidgetManager = AppWidgetManager.getInstance(context);
            ComponentName componentName = new ComponentName(context, GameWidget.class);
            int[] appWidgetIds = appWidgetManager.getAppWidgetIds(componentName);

            RemoteViews views = new RemoteViews(context.getPackageName(), R.layout.widget_game);

            appWidgetManager.updateAppWidget(appWidgetIds, views);
            appWidgetManager.notifyAppWidgetViewDataChanged(appWidgetIds, R.id.lv_games);
        }

        super.onReceive(context, intent);
    }

    public static void sendBroadcast(Context context) {
        Intent intentWidget = new Intent(AppWidgetManager.ACTION_APPWIDGET_UPDATE);
        intentWidget.setComponent(new ComponentName(context, GameWidget.class));
        context.sendBroadcast(intentWidget);
    }
}

File: app/src/main/java/com/cristhian/gametobeat/ui/widget/GameWidget.kt
package com.cristhian.gametobeat.ui.widget

import android.appwidget.AppWidgetManager
import android.appwidget.AppWidgetProvider
import android.content.ComponentName
import android.content.Context
import android.content.Intent
import android.widget.RemoteViews
import com.cristhian.gametobeat.R
import com.cristhian.gametobeat.database.Game

class GameWidget : AppWidgetProvider() {
    override fun onUpdate(
            context: Context,
            appWidgetManager: AppWidgetManager,
            appWidgetIds: IntArray
    ) { // There may be multiple widgets active, so update all of them
        for (appWidgetId in appWidgetIds) {
            updateAppWidget(context, appWidgetManager, appWidgetId)
        }
    }

    override fun onEnabled(context: Context) {}
    override fun onDisabled(context: Context) {}
    override fun onReceive(context: Context, intent: Intent) {
        val action = intent.action
        if (action == AppWidgetManager.ACTION_APPWIDGET_UPDATE) {
            val appWidgetManager = AppWidgetManager.getInstance(context)
            val componentName = ComponentName(context, GameWidget::class.java)
            val appWidgetIds = appWidgetManager.getAppWidgetIds(componentName)
            val views = RemoteViews(context.packageName, R.layout.widget_game)
            appWidgetManager.updateAppWidget(appWidgetIds, views)
            appWidgetManager.notifyAppWidgetViewDataChanged(appWidgetIds, R.id.lv_games)
        }
        super.onReceive(context, intent)
    }

    companion object {
        var mGames: List<Game>? = null
        fun updateAppWidget(
                context: Context, appWidgetManager: AppWidgetManager,
                appWidgetId: Int
        ) { // Construct the RemoteViews object
            val views = RemoteViews(context.packageName, R.layout.widget_game)
            val intentGames = Intent(context, GameWidgetService::class.java)
            views.setRemoteAdapter(R.id.lv_games, intentGames)
            // Instruct the widget manager to update the widget
            appWidgetManager.updateAppWidget(appWidgetId, views)
        }

        fun sendBroadcast(context: Context) {
            val intentWidget = Intent(AppWidgetManager.ACTION_APPWIDGET_UPDATE)
            intentWidget.component = ComponentName(context, GameWidget::class.java)
            context.sendBroadcast(intentWidget)
        }
    }
}